pytest_annotate/plugin.py

```diff
@@ -3,28 +3,39 @@
 
 import os
 
-from pyannotate_runtime import collect_types
-
 
 class PyAnnotatePlugin(object):
     """A pytest plugin that profiles function calls to extract type info."""
 
     def __init__(self, output_file):
         """Create a new PyAnnotatePlugin that analyzes function calls to extract type info."""
         self.output_file = output_file
-        collect_types.init_types_collection()
+        self.collect_types = None
+
+    def pytest_collection_finish(self, session):
+        """Handle the pytest collection finish hook: configure pyannotate.
+
+        Explicitly delay importing `collect_types` all tests have been collected.  This gives
+        gevent a chance to monkey patch the world before importing pyannotate.
+        """
+        from pyannotate_runtime import collect_types  # noqa
+        self.collect_types = collect_types
+        self.collect_types.init_types_collection()
 
     def pytest_unconfigure(self, config):
         """Unconfigure the pytest plugin. Happens when pytest is about to exit."""
-        collect_types.dump_stats(self.output_file)
+        if self.collect_types:
```

Comment:
Presumably `self.collect_types` should never be falsy in any of these hooks, since test collection should finish before any of them are called, no? Can we make the `else` case in each hook be `raise AssertionError` with an appropriate error message? (I dislike the `assert` keyword since it can be turned off, and `bandit` will complain about it as well)